sysdeps/linux/generic/sysdeps.cpp

```diff
@@ -625,14 +718,206 @@ int sys_reboot(int cmd) {
 	return 0;
 }
 
+int sys_getaffinity(pid_t pid, size_t cpusetsize, cpu_set_t *mask) {
+	auto ret = do_syscall(SYS_sched_getaffinity, pid, cpusetsize, mask);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_mount(const char *source, const char *target,
+	const char *fstype, unsigned long flags, const void *data) {
+	auto ret = do_syscall(SYS_mount, source, target, fstype, flags, data);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_umount2(const char *target, int flags) {
+	auto ret = do_syscall(SYS_umount2, target, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_sethostname(const char *buffer, size_t bufsize) {
+	auto ret = do_syscall(SYS_sethostname, buffer, bufsize);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_epoll_create(int flags, int *fd) {
+	auto ret = do_syscall(SYS_epoll_create1, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	*fd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_epoll_ctl(int epfd, int mode, int fd, struct epoll_event *ev) {
+	auto ret = do_syscall(SYS_epoll_ctl, epfd, mode, fd, ev);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_epoll_pwait(int epfd, struct epoll_event *ev, int n, int timeout, const sigset_t *sigmask, int *raised) {
+	auto ret = do_syscall(SYS_epoll_pwait, epfd, ev, n, timeout, sigmask, NSIG / 8);
+	if (int e = sc_error(ret); e)
+		return e;
+	*raised = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_eventfd_create(unsigned int initval, int flags, int *fd) {
+	auto ret = do_syscall(SYS_eventfd2, initval, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	*fd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_signalfd_create(const sigset_t *masks, int flags, int *fd) {
+	auto ret = do_syscall(SYS_signalfd4, *fd, masks, 65/8, flags);
```

Comment:
Did you mean to use `NSIG` here?

Reply:
Is that even correct? Why not `sizeof(sigset_t)`?

Reply:
As far as the (Linux) kernel is concerned, it can handle `sizemask` not being equal to `sizeof sigset_t` (see the `SYSCALL_DEFINE4` for `signalfd4` in signalfd.c). Passing the supported count of signals seems reasonable here. Cross-checking with musl, they do it using NSIG, too.

Reply:
A fixup has already been pushed changing this to utilize NSIG.